src/vmm/src/vstate/vcpu.rs

```diff
@@ -661,12 +632,13 @@ impl VcpuHandle {
     /// # Errors
     ///
     /// When [`vmm_sys_util::linux::signal::Killable::kill`] errors.
-    pub fn send_event(&self, event: VcpuEvent) -> Result<(), VcpuSendEventError> {
+    pub fn send_event(&mut self, event: VcpuEvent) -> Result<(), VcpuSendEventError> {
         // Use expect() to crash if the other thread closed this channel.
         self.event_sender
             .send(event)
             .expect("event sender channel closed on vcpu end.");
         // Kick the vcpu so it picks up the message.
+        self.vcpu_fd.set_kvm_immediate_exit(1);
```

Comment:
mh. I wonder if we should have a `fence(Release)` here, because we must make sure that by the time we send the signal, the vCPU thread sees the updated value for `immediate_exit`.

Reply:
+1

Reply:
added

Reply:
added